### DISCUSSION TITLE
running time is too long when dimension reduce for merged 8 samples

### DISCUSSION THREAD

**Discussion by xiaowenchenjax** [stuart-lab/signac#1000] [Q&A] [answered]
Dear developer,
I have 8 samples, i have merged them together starting from "fragments.tsv.gz" files. the merged object has 245020 features, and 2809084 samples. I have two questions
1. I am running "Dimension reduction : runSVD", which takes a long time, about five days, it still running. what i should do?
2. How I started the 8 samples which have been removed low cells to merge them?

Thank you.

**Comment by timoast** [comment, marked as answer]
I have not processed datasets that large (>2.8 million cells), so I can't comment on how long I would expect it to take. However, we provide some benchmarks in the [Signac paper](https://www.nature.com/articles/s41592-021-01282-5) (Figure 5) showing that 600k cells takes around 80 min, and I would expect a runtime of around 8 hours for 3M cells extrapolating linearly from our benchmarks. This would also depend on the number of features and the computational hardware. To speed things up, I'd suggest reducing the number of features if you can (eg, set a higher threshold for `min.cutoff` in `FindTopFeatures()` or refine the peak calling), and try increasing the `irlba.work` parameter in `RunSVD()` (you can try setting this a lot higher if you have memory available).

> I am running "Dimension reduction : runSVD"

Which vignette are you referring to?

> How I started the 8 samples which have been removed low cells to merge them?

Not sure what you mean exactly. If you want to merge datasets, you can take a look at the [merge vignette](https://satijalab.org/signac/articles/merging.html)

**Reply by xiaowenchenjax** [reply]
Sorry, I did not describe it clearly. 
for the second question, i mean that how I merge all the 8 samples after filtering the low quality cells for each samples based on QC, since the merge vignette tell us how to merge all the 8 samples using the output of cellranger.

**Reply by timoast** [reply]
The workflow is the same whether you want to remove low quality cells before or after merging. The main point is that you need to quantify the same features across all the objects you need to merge, and then use the `merge()` function

**Reply by xiaowenchenjax** [reply]
Got it. Thanks for your quick reply.
